python/filtering/filter_menu.py

```diff
@@ -264,6 +268,9 @@ def __init__(
         # based on menu selection.
         self._active_filter = FilterItem.create_group(FilterItem.FilterOp.AND)
 
+        # Set the active preset filter name to None, indicating no preset filter is active.
+        self._active_preset_filter_name = None
```

Comment:
Is there a reason to treat preset filters differently than any other filters in the menu?

I would rather include the preset filter as part of the `self._active_filter`, this way all filters are contained in one object and filtering is handled by the framework (instead of having each App implement its own preset filter handling, which will simplify your tk-multi-shotgunpanel PR).

You should be able to modify `get_current_filters` to append the preset filter to the list of filters, which will create an AND with all other filters currently applied, and all filtering will be handled for you by the framework

Reply:
hey @staceyoue !

I'm curious to see how you would handle these filters from a front-end perspective? It leads me to some questions :)

- how/where would you define them? still in a hook/setting by app I guess?
- then, you would modify the load method of the model(s) in each app to make sure all the fields required by the filters are fetched from FPTR when querying for the entities?
- what about if the applied preset is using a condition on a field and the same field is used by the user in the filter menu? does the `get_current_filters` method (or another one) should handle priorities? (for example, first apply the preset filter, then apply the other filters?)
- or should the preset filter be applied and then the list of other filters be refreshed accordingly? I don't remember if they're managing the "blank" value for example

Reply:
All good points!

So after digging into this more, the TLDR: because the preset filters need to be defined as "shotgun filters", then they do need to be backend-filters.

It would be nice to have the framework handle all the filtering functionality, instead of each App that wants preset filters to have to introduce their own logic to handle the filtering, but currently the framework filtering can only handle filters that are built from the model data (and not shotgun filters).